FORGE_DB_NAME breaks if passed a branch name with a hypen

# Description

Databases specified with the FORGE_DB_NAME variable that include hyphens break the Create Database pipeline.

# Example

Setting this environment variable in the Github Action: 

FORGE_SITE_BRANCH: ${{ github.head_ref }}

 on a branch called `ABC-123` would result in it trying to create a database named `ABC-123` which is invalid: 
 
 ```
  INFO    Start finding the server.
  INFO    Finding the associated site.
  INFO    ---> Site not found.
  INFO    Creating a new site.
  INFO    ---> Use the specified Nginx template.
  INFO    Creating database.
  FAIL    ---> The name must only contain letters, numbers, and underscores.

   Laravel\Forge\Exceptions\ValidationException 

  The given data failed to pass validation.

  at /composer/vendor/laravel/forge-sdk/src/MakesHttpRequests.php:106
    102▕      */
    103▕     protected function handleRequestError(ResponseInterface $response)
    104▕     {
    105▕         if ($response->getStatusCode() == 422) {
  ➜ 106▕             throw new ValidationException(json_decode((string) $response->getBody(), true));
    107▕         }
    108▕ 
    109▕         if ($response->getStatusCode() == 404) {
    110▕             throw new NotFoundException();


Error: Process completed with exit code 1.

```

# Fix

Replace hyphens with underscores when setting the database name in the Forge Settings initialisation.

--- a/app/Providers/AppServiceProvider.php
+++ b/app/Providers/AppServiceProvider.php
@@ -11,9 +11,7 @@ class AppServiceProvider extends ServiceProvider
 {
     public function boot(): void
     {
-        $this->app->setLocale('en');
-
-        $this->loadTranslationsFrom(base_path('lang'), 'harbor');
+        //
     }
 
     public function register(): void

--- a/app/Services/Forge/ForgeService.php
+++ b/app/Services/Forge/ForgeService.php
@@ -80,7 +80,7 @@ class ForgeService
         );
     }
 
-    public function getStandardizedBranchName(): string
+    public function getSiteIsolationUsername(): string
     {
         return GenerateStandardizedBranchName::run(
             $this->getFormattedBranchName()

--- a/app/Services/Forge/ForgeService.php
+++ b/app/Services/Forge/ForgeService.php
@@ -89,8 +89,8 @@ class ForgeService
 
     public function getFormattedDatabaseName(): string
     {
-        return GenerateStandardizedBranchName::run(
-            $this->setting->dbName ?? $this->getFormattedBranchName()
+        return $this->setting->dbName ?? GenerateStandardizedBranchName::run(
+            $this->getFormattedBranchName()
         );
     }
 

--- a/app/Services/Forge/ForgeService.php
+++ b/app/Services/Forge/ForgeService.php
@@ -87,6 +87,13 @@ class ForgeService
         );
     }
 
+    public function getFormattedDatabaseName(): string
+    {
+        return GenerateStandardizedBranchName::run(
+            $this->setting->dbName ?? $this->getFormattedBranchName()
+        );
+    }
+
     public function siteNginxTemplate(): string
     {
         return $this->forge->siteNginxFile($this->setting->server, $this->site->id);

--- a/app/Services/Forge/ForgeSetting.php
+++ b/app/Services/Forge/ForgeSetting.php
@@ -201,7 +201,7 @@ class ForgeSetting
         'site_isolation_required' => ['boolean'],
         'job_scheduler_required' => ['boolean'],
         'db_creation_required' => ['boolean'],
-        'db_name' => ['nullable', 'string'],
+        'db_name' => ['nullable', 'string', 'regex:/^[a-zA-Z0-9_]+$/'],
         'auto_source_required' => ['boolean'],
         'ssl_required' => ['boolean'],
         'wait_on_ssl' => ['boolean'],
@@ -224,7 +224,14 @@ class ForgeSetting
 
     private function init(array $configurations): void
     {
-        $validator = Validator::make($configurations, $this->validationRules);
+        $validator = Validator::make(
+            $configurations,
+            $this->validationRules,
+            [
+                'db_name.regex' => 'The :attribute must only contain letters, numbers, and underscores.'
+            ],
+            ['db_name' => 'Database name']
+        );
 
         throw_if($validator->fails(), ValidationException::class, $validator->errors()->all());
 

--- a/app/Services/Forge/ForgeSetting.php
+++ b/app/Services/Forge/ForgeSetting.php
@@ -230,7 +230,7 @@ class ForgeSetting
             [
                 'db_name.regex' => 'The :attribute must only contain letters, numbers, and underscores.'
             ],
-            ['db_name' => 'Database name']
+            ['db_name' => 'FORGE_DB_NAME']
         );
 
         throw_if($validator->fails(), ValidationException::class, $validator->errors()->all());

--- a/app/Services/Forge/Pipeline/CreateDatabase.php
+++ b/app/Services/Forge/Pipeline/CreateDatabase.php
@@ -13,6 +13,7 @@ declare(strict_types=1);
 
 namespace App\Services\Forge\Pipeline;
 
+use App\Actions\FormattedBranchName;
 use App\Services\Forge\ForgeService;
 use App\Traits\Outputifier;
 use Closure;
@@ -28,8 +29,13 @@ class CreateDatabase
             return $next($service);
         }
 
+        if ( $service->setting->dbName ) {
+            $dbName = FormattedBranchName::run($service->setting->dbName);
+        } else {
+            $dbName = $service->getStandardizedBranchName();
+        }
+
         $dbPassword = Str::random(16);
-        $dbName = $service->setting->dbName ?? $service->getStandardizedBranchName();
 
         if (! $this->databaseExists($service, $dbName)) {
             $this->information('Creating database.');

--- a/app/Services/Forge/Pipeline/CreateDatabase.php
+++ b/app/Services/Forge/Pipeline/CreateDatabase.php
@@ -29,12 +29,7 @@ class CreateDatabase
             return $next($service);
         }
 
-        if ( $service->setting->dbName ) {
-            $dbName = GenerateStandardizedBranchName::run($service->setting->dbName);
-        } else {
-            $dbName = $service->getStandardizedBranchName();
-        }
-
+        $dbName = $service->getFormattedDatabaseName();
         $dbPassword = Str::random(16);
 
         if (! $this->databaseExists($service, $dbName)) {

--- a/app/Services/Forge/Pipeline/CreateDatabase.php
+++ b/app/Services/Forge/Pipeline/CreateDatabase.php
@@ -13,7 +13,7 @@ declare(strict_types=1);
 
 namespace App\Services\Forge\Pipeline;
 
-use App\Actions\FormattedBranchName;
+use App\Actions\GenerateStandardizedBranchName;
 use App\Services\Forge\ForgeService;
 use App\Traits\Outputifier;
 use Closure;
@@ -30,7 +30,7 @@ class CreateDatabase
         }
 
         if ( $service->setting->dbName ) {
-            $dbName = FormattedBranchName::run($service->setting->dbName);
+            $dbName = GenerateStandardizedBranchName::run($service->setting->dbName);
         } else {
             $dbName = $service->getStandardizedBranchName();
         }

--- a/app/Services/Forge/Pipeline/OrCreateNewSite.php
+++ b/app/Services/Forge/Pipeline/OrCreateNewSite.php
@@ -54,7 +54,7 @@ class OrCreateNewSite
             $this->information('---> Enabling site isolation.');
 
             $data['isolated'] = true;
-            $data['username'] = $service->getStandardizedBranchName();
+            $data['username'] = $service->getSiteIsolationUsername();
         }
 
         return $data;

--- a/app/Services/Forge/Pipeline/RemoveDatabaseUser.php
+++ b/app/Services/Forge/Pipeline/RemoveDatabaseUser.php
@@ -24,7 +24,7 @@ class RemoveDatabaseUser
     public function __invoke(ForgeService $service, Closure $next)
     {
         foreach ($service->forge->databaseUsers($service->setting->server) as $databaseUser) {
-            if ($databaseUser->name === $service->getStandardizedBranchName()) {
+            if ($databaseUser->name === $service->getFormattedDatabaseName()) {
                 $this->information('Removing database with user.');
 
                 foreach ($databaseUser->databases as $database) {

--- a/config/app.php
+++ b/config/app.php
@@ -57,6 +57,18 @@ return [
 
     /*
     |--------------------------------------------------------------------------
+    | Application Locale Configuration
+    |--------------------------------------------------------------------------
+    |
+    | The application locale determines the default locale that will be used
+    | by the translation service provider. You are free to set this value
+    | to any of the locales which will be supported by the application.
+    |
+    */
+    'locale' => 'en',
+
+    /*
+    |--------------------------------------------------------------------------
     | Autoloaded Service Providers
     |--------------------------------------------------------------------------
     |

--- a/resources/lang/en/valiadtion.php
+++ b/resources/lang/en/valiadtion.php
@@ -180,7 +180,7 @@ return [
     */
 
     'attributes' => [
-        'db_name' => 'Database name'
+        'db_name' => 'FORGE_DB_NAME'
     ],
 
 ];

--- a/resources/lang/en/valiadtion.php
+++ b/resources/lang/en/valiadtion.php
@@ -163,8 +163,8 @@ return [
     */
 
     'custom' => [
-        'attribute-name' => [
-            'rule-name' => 'custom-message',
+        'db_name' => [
+            'regex' => 'The :attribute must only contain letters, numbers, and underscores.',
         ],
     ],
 
@@ -179,6 +179,8 @@ return [
     |
     */
 
-    'attributes' => [],
+    'attributes' => [
+        'db_name' => 'Database name'
+    ],
 
 ];

